sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -99,6 +103,10 @@ public class ChatClient {
         )
 
         self.service = client.chat
+        self.pushNotificationClient = PushNotificationClient(
+            credential: credential,
+            options: options
+        )
```

Comment:
What if the user does not want or care about push notifications? Why are we creating a push notification client? If creating the client fails for some reason it means that all ChatClient functionality is closed to users.

Reply:
Good question. Android initializes the PN client inside the Chat Client initialization process while RT Notification doesn't. Since PN and RT notification are in the same SDK. I think it is better to expect PN client has the same behaviour as RT client. So I will set PN Client as optional as well and initialize it when the contuse calls the related APIs.